src/xml_parsing.cpp

```diff
@@ -536,6 +536,25 @@ void VerifyXML(const std::string& xml_text,
       }
       else if(search->second == NodeType::CONTROL)
       {
+        if (name == "Sequence" && children_count <= 2)
+        {
+          ThrowError(line_number,
+                     std::string("A Sequence node must have 2 or more children"));
+        }
+        else if (name == "ReactiveSequence" && children_count <= 2)
+        {
+          ThrowError(line_number,
+                     std::string("A ReactiveSequence node must have 2 or more children"));
+        }
+        if (name == "ReactiveSequence")
+        {
+          auto first_child_node = search->second->FirstChildElement();
+          if (first_child_node != NodeType::CONDITION)
+          {
+            ThrowError(line_number,
+                      std::string("The first child of a ReactiveSequence should be of Condition type"));
```

Comment:
The first child of a ReactiveSequence should be a Condition that returns SUCCESS or FAILURE immediately.